### ISSUE TITLE
"compiler produced stdout" stat undocumented.

### ISSUE THREAD

**Issue by phraemer** [ccache/ccache#142]
Hi,

In the stats there is a line "compiler produced stdout"
I can find no explanation in the documentation or elsewhere of what it means exactly and what should be done to avoid this.

thanks!
James

**Comment by afbjorklund** [comment]
Normally the compiler produces output in the provided object file (and friends, like dependency files), and warnings/errors on stderr. If there is anything on stdout, then something is "broken" inside the compiler.

**Comment by phraemer** [comment]
I see. I my case I'm using `distcc` in pump mode and it's normal for pump to output to stdout.

e.g.
```bash
ccache pump distcc clang -v -c main.c -o main.o
```

This will output the following to stdout...
```
__________Using distcc-pump from /usr/local/bin/../Cellar/distcc/3.2rc1/bin
__________Using        1 distcc server in pump mode
__________Shutting down distcc-pump include server
```

ccache will then report "compiler produced stdout", fallback to the compiler.

Is this really a problem with pump or should ccache accept output to stdout, or at least be configurable?

**Comment by afbjorklund** [comment]
I think there is an integration problem... You also really want the "cpp2" mode (since pump mode doesn't handle preprocessed source code very well), but since that is now the default (since 3.3) it _should_ work.

i.e. we should improve on the current "state of affairs" (from distcc's doc):

> Use with ccache
> ---------------
> 
> The best way to use is to set up a similar masquerade directory for
> ccache and put it on the PATH before distcc.
> 
> NOTE: This use of ccache is incompatible with use of distcc's "pump" mode.
> 
> (If you're using "pump" mode, it might be possible to use ccache on
> the distcc server machines. But we haven't tried that setup.)
> 

Using distcc in non-pump (i.e. normal) mode should work OK with ccache 3.1-3.2, once you get rid of some of all the bugs with configuration in those old distcc versions... Hopefully distcc 3.3 will be better.

If you decide to use ccache only on the distcc servers, then you really want those to share a cache...

----
Ultimately, one would like both the cache server and the include server to be started up *once*. Then _all_ the compilations for the current build would be performed against those, instead of stopping/starting.

Basically the cache server (manifest memcached) only stores the timestamps and checksums of files, whereas the include server (distcc include_server.py) also stores the compressed contents of them.

But that's all work-in-progress...

**Comment by afbjorklund** [comment]
@phraemer : I made some adjustments in 9327d05, so that you can use it with `distcc-pump` too. Basically just ignoring those silly lines on standard output, with CCACHE_PREFIX="pump distcc".

The recommended way is (the default) `run_second_cpp=true`, otherwise you will get a warning like:
`Warning: cannot use distcc_pump on already preprocessed file (such as emitted by ccache)`

Will do a separate Pull Request.

**Comment by phraemer** [comment]
Great! Thank you!
